roxie/ccd/ccdserver.cpp

```diff
@@ -5207,14 +5216,18 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
                     }
                 }
             }
-            else
+            else if (!localAgent)
```

Comment:
Why have an else and possibly do the retry only in the else clause - why not always check at a regular interval (implies not calling msTick() so often but some sort of timer/poll) and then if there are lots of other msgs/activity retries are still checked at a regular interval ?

Reply:
In the acknowledgeAll case, I do... Leaving the old code closer to how it used to behave simply because it's only there as a fall-back in case of unanticipated issues in the new...

Reply:
Maybe I am reading the code/loop incorrectly and if so I apologize, but I think I see -
```
for(;;)
{
    mr = ...
    if (mr)
    {
       ...
    }
    else if (!localAgent)
    {
        if(acknowledgeAllRequests)
            retryPending(checkInterval)
    }
}
```
and perhaps its just not an issue, but if handing msgs often (mr != nullptr) then the else that calls retryPending() is not called at some minimum guaranteed interval ?

Reply:
The
if (mr)
code path can return if there are continuations?, so if we do make any changes to this loop, perhaps the
retryPending(checkInterval) 
call should be made before the if(mr) check ?
Anyway let me know if this sort of change is worth any more consideration.